keras/backend/theano_backend.py

```diff
@@ -998,9 +998,22 @@ def _step(input, *states):
 
 
 def switch(condition, then_expression, else_expression):
-    '''condition: scalar tensor.
+    '''Switches between two operations depending on `condition`.
+    Note that both `then_expression` and `else_expression`
+    should be symbolic tensors of the *same shape*.
+
+    # Arguments
+        condition: a scalar, or tensor whose shape compatible with then_expression.
+        then_expression: TensorFlow operation.
+        else_expression: TensorFlow operation.
     '''
-    return T.switch(condition, then_expression, else_expression)
+    if ndim(condition) == 0:
+        return theano.ifelse.ifelse(condition, then_expression, else_expression)
+    else:
+        dims = ndim(then_expression)
+        while(ndim(condition) < dims):
+            condition = expand_dims(condition)
```

Comment:
Is this really useful? It might be more confusing than helpful, because it's making the behaviour of `K.switch` different from that of `theano.tensor.switch`.